### ISSUE TITLE
ImportError: libgnuradio-lora_sdr.so.0.5.0git: cannot open shared object file: No such file or directory (Ubuntu Jammy)

### ISSUE THREAD

**Issue by vogelfreiheit** [martynvdijke/gr-lora_sdr#30]
I have built the sources properly and installed in /usr/local. Any attempt to use companion or the standalone python files in the affected Ubuntu Jammy system yields this error:

`ImportError: libgnuradio-lora_sdr.so.0.5.0git: cannot open shared object file: No such file or directory`

```
ldd /usr/local/lib/python3.10/dist-packages/lora_sdr/lora_sdr_python.cpython-310-x86_64-linux-gnu.so
	linux-vdso.so.1 (0x00007ffd5c3bd000)
	libgnuradio-lora_sdr.so.0.5.0git => not found
	libgnuradio-runtime.so.3.10.1 => /lib/x86_64-linux-gnu/libgnuradio-runtime.so.3.10.1 (0x00007f9b54850000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f9b54624000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f9b54604000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f9b543dc000)
	libgnuradio-pmt.so.3.10.1 => /lib/x86_64-linux-gnu/libgnuradio-pmt.so.3.10.1 (0x00007f9b5437f000)
	libvolk.so.2.5 => /lib/x86_64-linux-gnu/libvolk.so.2.5 (0x00007f9b540ba000)
	libboost_program_options.so.1.74.0 => /lib/x86_64-linux-gnu/libboost_program_options.so.1.74.0 (0x00007f9b54075000)
	libboost_thread.so.1.74.0 => /lib/x86_64-linux-gnu/libboost_thread.so.1.74.0 (0x00007f9b54053000)
	libspdlog.so.1 => /lib/x86_64-linux-gnu/libspdlog.so.1 (0x00007f9b53fd8000)
	libfmt.so.8 => /lib/x86_64-linux-gnu/libfmt.so.8 (0x00007f9b53fb7000)
	libgmp.so.10 => /lib/x86_64-linux-gnu/libgmp.so.10 (0x00007f9b53f33000)
	libunwind.so.8 => /lib/x86_64-linux-gnu/libunwind.so.8 (0x00007f9b53f18000)
	libpython3.10.so.1.0 => /lib/x86_64-linux-gnu/libpython3.10.so.1.0 (0x00007f9b53940000)
	libthrift-0.16.0.so => /lib/x86_64-linux-gnu/libthrift-0.16.0.so (0x00007f9b538a7000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f9b537c0000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f9b54a46000)
	liborc-0.4.so.0 => /lib/x86_64-linux-gnu/liborc-0.4.so.0 (0x00007f9b53739000)
	liblzma.so.5 => /lib/x86_64-linux-gnu/liblzma.so.5 (0x00007f9b5370e000)
	libexpat.so.1 => /lib/x86_64-linux-gnu/libexpat.so.1 (0x00007f9b536dd000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f9b536c1000)
	libssl.so.3 => /lib/x86_64-linux-gnu/libssl.so.3 (0x00007f9b5361d000)
	libcrypto.so.3 => /lib/x86_64-linux-gnu/libcrypto.so.3 (0x00007f9b531d9000)
```
```
$ LD_LIBRARY_PATH=/usr/local/lib python apps/single_user/tx_rx_simulation.py 
Traceback (most recent call last):
  File "/home/user/code/thirdparty/gr-lora_sdr/apps/single_user/tx_rx_simulation.py", line 22, in <module>
    import lora_sdr
  File "/usr/local/lib/python3.10/dist-packages/lora_sdr/__init__.py", line 18, in <module>
    from .lora_sdr_python import *
ImportError: libgnuradio-lora_sdr.so.0.5.0git: cannot open shared object file: No such file or directory
```
```
$ find /usr/local/ -name libgnuradio-lora_sdr.so.0.5.0git
/usr/local/lib/x86_64-linux-gnu/libgnuradio-lora_sdr.so.0.5.0git
```

```
$ ldd /usr/local/lib/x86_64-linux-gnu/libgnuradio-lora_sdr.so.0.5.0git
	linux-vdso.so.1 (0x00007ffe37953000)
	libgnuradio-runtime.so.3.10.1 => /lib/x86_64-linux-gnu/libgnuradio-runtime.so.3.10.1 (0x00007f6ef76a4000)
	libgnuradio-pmt.so.3.10.1 => /lib/x86_64-linux-gnu/libgnuradio-pmt.so.3.10.1 (0x00007f6ef7649000)
	libboost_thread.so.1.74.0 => /lib/x86_64-linux-gnu/libboost_thread.so.1.74.0 (0x00007f6ef7627000)
	libvolk.so.2.5 => /lib/x86_64-linux-gnu/libvolk.so.2.5 (0x00007f6ef7362000)
	libspdlog.so.1 => /lib/x86_64-linux-gnu/libspdlog.so.1 (0x00007f6ef72e7000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f6ef70b9000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f6ef6fd2000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f6ef6fb2000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f6ef6d8a000)
	libboost_program_options.so.1.74.0 => /lib/x86_64-linux-gnu/libboost_program_options.so.1.74.0 (0x00007f6ef6d45000)
	libfmt.so.8 => /lib/x86_64-linux-gnu/libfmt.so.8 (0x00007f6ef6d24000)
	libgmp.so.10 => /lib/x86_64-linux-gnu/libgmp.so.10 (0x00007f6ef6ca0000)
	libunwind.so.8 => /lib/x86_64-linux-gnu/libunwind.so.8 (0x00007f6ef6c85000)
	libpython3.10.so.1.0 => /lib/x86_64-linux-gnu/libpython3.10.so.1.0 (0x00007f6ef66ad000)
	libthrift-0.16.0.so => /lib/x86_64-linux-gnu/libthrift-0.16.0.so (0x00007f6ef6614000)
	liborc-0.4.so.0 => /lib/x86_64-linux-gnu/liborc-0.4.so.0 (0x00007f6ef658f000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f6ef78fd000)
	liblzma.so.5 => /lib/x86_64-linux-gnu/liblzma.so.5 (0x00007f6ef6562000)
	libexpat.so.1 => /lib/x86_64-linux-gnu/libexpat.so.1 (0x00007f6ef6531000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f6ef6515000)
	libssl.so.3 => /lib/x86_64-linux-gnu/libssl.so.3 (0x00007f6ef6471000)
	libcrypto.so.3 => /lib/x86_64-linux-gnu/libcrypto.so.3 (0x00007f6ef602f000)
```

**Comment by vogelfreiheit** [comment]
Symlink seems fine:

```
ls -l /usr/local/lib/x86_64-linux-gnu/libgnuradio-lora_sdr.so.0.5.0git
lrwxrwxrwx 1 root root 32 may  8 17:34 /usr/local/lib/x86_64-linux-gnu/libgnuradio-lora_sdr.so.0.5.0git -> libgnuradio-lora_sdr.so.af08300e
```

**Comment by vogelfreiheit** [comment]
A quick test forcing the LD path:

```
$ LD_LIBRARY_PATH=/usr/local/lib/x86_64-linux-gnu python apps/single_user/tx_rx_simulation.py 
Traceback (most recent call last):
  File "/home/user/code/thirdparty/gr-lora_sdr/apps/single_user/tx_rx_simulation.py", line 188, in <module>
    main()
  File "/home/user/code/thirdparty/gr-lora_sdr/apps/single_user/tx_rx_simulation.py", line 166, in main
    tb = top_block_cls()
  File "/home/user/code/thirdparty/gr-lora_sdr/apps/single_user/tx_rx_simulation.py", line 49, in __init__
    self.lora_sdr_modulate_0 = lora_sdr.modulate(sf, samp_rate, bw, [8, 16],False)
AttributeError: module 'lora_sdr' has no attribute 'modulate'
```

Seems like a SWIG issue. modulate() is clearly missing from the Py module.

**Comment by vogelfreiheit** [comment]
See https://github.com/martynvdijke/gr-lora_sdr/issues/27

**Comment by vogelfreiheit** [comment]
cmake output:
```
-- The CXX compiler identification is GNU 11.2.0
-- The C compiler identification is GNU 11.2.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Found setting GRLORA_MEASUREMENTS=True, setting outputting meassurments on
-- Build type not specified: defaulting to release.
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.2") 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Checking for module 'gmp'
--   Found gmp, version 6.2.1
-- Found GMP: /usr/lib/x86_64-linux-gnu/libgmpxx.so  
-- Using GMP.
-- Found MPLIB: /usr/lib/x86_64-linux-gnu/libgmpxx.so  
-- Found Boost: /usr/lib/x86_64-linux-gnu/cmake/Boost-1.74.0/BoostConfig.cmake (found suitable version "1.74.0", minimum required is "1.74.0") found components: date_time program_options system regex thread unit_test_framework 
-- Found Volk: Volk::volk  
-- User set python executable /usr/bin/python3
-- Found PythonInterp: /usr/bin/python3 (found version "3.10.4") 
-- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython3.10.so (found suitable exact version "3.10.4") 
-- Found Git: /usr/bin/git  
-- Extracting version information from git describe...
-- Found Doxygen: /usr/bin/doxygen (found version "1.9.1") found components: doxygen dot 
-- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython3.10.so
-- Performing Test HAS_FLTO
-- Performing Test HAS_FLTO - Success
-- Found pybind11: /usr/include (found version "2.9.1")
-- Using install prefix: /usr
-- Building for version: af08300e / 0.5.0git
-- PYTHON and GRC components are enabled
-- Python checking for pygccxml - found
-- Configuring done
-- Generating done
-- Build files have been written to: /home/user/code/thirdparty/gr-lora_sdr/build
```

**Comment by vogelfreiheit** [comment]
Fixed, and ported bindings to 3.10.
